tests/test-cases/Group23-VIC-Machine-Service/23-03-VCH-Create.robot

```diff
@@ -277,3 +311,276 @@ Fail to create VCH with gateway without static address
+
+    Verify Return Code
+    Verify Status Created
+
+    Get Docker Host Params    vch    %{VCH-NAME}
+
+    Wait Until Keyword Succeeds    10x    2s    Pull Busy Box    ${docker_host} --tls    1    Access denied to unauthorized registry
+
+    [Teardown]    Run Secret VIC Machine Delete Command    %{VCH-NAME}
+
+
+Fail to create VCH where insecure property contains invalid char - registry settings
+    Create VCH    '{"name":"%{VCH-NAME}-invalid_registry","compute":{"resource":{"name":"%{TEST_RESOURCE}"}},"storage":{"image_stores":["ds://%{TEST_DATASTORE}"]},"network":{"bridge":{"ip_range":"172.16.0.0/12","port_group":{"name":"%{BRIDGE_NETWORK}"}},"public":{"port_group":{"name":"${PUBLIC_NETWORK}"}}},"registry":{"image_fetch_proxy":{"http":"http://example.com","https":"https://example.com"},"insecure":[https://insecure.example.com],"whitelist":["10.0.0.0/8"]},"auth":{"server":{"generate":{"cname":"vch.example.com","organization":["VMware, Inc."],"size":{"value":2048,"units":"bits"}}},"client":{"no_tls_verify": true}}}'
+
+    Verify Return Code
+    Verify Status Bad Request
+
+    Output Should Contain    failed, because invalid character
+
+
+Fail to create VCH where insecure property contains an int in one of the string arrays - registry settings
+    Create VCH    '{"name":"%{VCH-NAME}-invalid_registry","compute":{"resource":{"name":"%{TEST_RESOURCE}"}},"storage":{"image_stores":["ds://%{TEST_DATASTORE}"]},"network":{"bridge":{"ip_range":"172.16.0.0/12","port_group":{"name":"%{BRIDGE_NETWORK}"}},"public":{"port_group":{"name":"${PUBLIC_NETWORK}"}}},"registry":{"image_fetch_proxy":{"http":"http://example.com","https":"https://example.com"},"insecure":["https://insecure.example.com", 101010101010, "https://insecure.example.com"],"whitelist":["10.0.0.0/8"]},"auth":{"server":{"generate":{"cname":"vch.example.com","organization":["VMware, Inc."],"size":{"value":2048,"units":"bits"}}},"client":{"no_tls_verify": true}}}'
+
+    Verify Return Code
+    Verify Status Bad Request
+
+    Output Should Contain    failed, because json: cannot unmarshal number into Go struct field
+
+
+Fail to create VCH where compute resource property contains invalid data - resource settings
+    Create VCH    '{"name":"%{VCH-NAME}-invalid_registry","compute":{"resource":{"name":"TEST_RESOURCE"}},"storage":{"image_stores":["ds://%{TEST_DATASTORE}"]},"network":{"bridge":{"ip_range":"172.16.0.0/12","port_group":{"name":"%{BRIDGE_NETWORK}"}},"public":{"port_group":{"name":"${PUBLIC_NETWORK}"}}},"registry":{"image_fetch_proxy":{"http":"http://example.com","https":"https://example.com"},"insecure":["https://insecure.example.com"],"whitelist":["10.0.0.0/8"]},"auth":{"server":{"generate":{"cname":"vch.example.com","organization":["VMware, Inc."],"size":{"value":2048,"units":"bits"}}},"client":{"no_tls_verify": true}}}'
+
+    Verify Return Code
+    Verify Status Bad Request
+
+    Output Should Contain    not found
```

Comment:
I followed the pattern for the rest of the test cases in this suite and used specific/shorten parts of the error message for verification.  The actual error message is more helpful/descriptive.

Reply:
> pattern for the rest of the test cases in this suite and used specific/shorten parts of the error message for verification

For what it's worth, I tried to capture the "essence" of the error message in the other checks: something that should always be present, even if the exact wording/formatting of the error changes (since it's always annoying to upgrade a library or improve an error message only to have a ton of tests fail that were depending on the exact wording of the error).

I think checking for "not found" fits with this pattern. Also checking that the output contains `TEST_RESOURCE` (the thing we can't find) seems like a reasonable way to make sure the error message is at least _moderately_ useful, without making the test fragile.

Reply:
I'll update this to check for `TEST_RESOURCE` as well.